source/extensions/network/dns_resolver/cares/dns_impl.cc

```diff
@@ -232,6 +232,9 @@ void DnsResolverImpl::PendingResolution::finishResolve() {
       ENVOY_LOG(critical, "Unknown exception in c-ares callback");
       dispatcher_.post([] { throw EnvoyException("unknown"); });
     }
+  } else {
+    ENVOY_LOG_EVENT(debug, "cares_dns_callback_cancelled",
+                    "dns resolution callback for {} not issued", dns_name_);
```

Comment:
May want to store and print out the `CancelReason` when [`cancel()` is called](https://github.com/envoyproxy/envoy/blob/4e5fb4e01d02ee9acb9bd13e124b865e1cabef91/source/extensions/network/dns_resolver/cares/dns_impl.h#L43) as it may be more informative.